src/aux-server/server/modules/GpioModule3.ts

```diff
@@ -0,0 +1,491 @@
+import { AuxModule2, Simulation } from '@casual-simulation/aux-vm';
+import {
+    DeviceInfo,
+    remoteResult,
+    remoteError,
+} from '@casual-simulation/causal-trees';
+import { Subscription } from 'rxjs';
+import { flatMap } from 'rxjs/operators';
+import {
+    SerialConnectAction,
+    SerialStreamAction,
+    SerialOpenAction,
+    SerialUpdateAction,
+    SerialWriteAction,
+    SerialReadAction,
+    SerialCloseAction,
+    // SerialSetAction,
+    // SerialGetAction,
+    // SerialFlushAction,
+    // SerialDrainAction,
+    SerialPauseAction,
+    SerialResumeAction,
+    asyncResult,
+    asyncError,
+    hasValue,
+} from '@casual-simulation/aux-common';
+import { Callback } from 'redis';
+const execSync = require('child_process').execSync;
+const SerialPort = require('serialport');
+const parsers = SerialPort.parsers;
+
+let btSerial = new Map<string, typeof SerialPort>();
+
+/**
+ * Defines an AuxModule that adds Serial functionality to the module.
+ */
+export class GpioModule3 implements AuxModule2 {
```

Comment:
@WesleyMayle maybe this class & file should be named `SerialModule` because it uses serial ports instead of GPIO stuff?

Reply:
Yeah, I can rename it quick